lib/ABI/FunctionType.cpp

```diff
@@ -278,6 +285,36 @@ class ConvertionHelper {
     return Result;
   }
 
+  static InternalLayout getLayout(const model::CABIFunctionType &Function) {
+    InternalLayout Result;
+
+    size_t CurrentOffset = 0;
+    DistributedArguments Arguments = distributeArguments(Function.Arguments);
+    revng_assert(Arguments.size() == Function.Arguments.size());
+    for (size_t Index = 0; Index < Arguments.size(); ++Index) {
+      auto &Current = Result.Arguments.emplace_back();
+      Current.Registers = std::move(Arguments[Index].Registers);
+      if (Arguments[Index].SizeOnStack != 0) {
+        /// \todo maybe some kind of alignment considerations are needed here.
+        Current.Stack = typename Layout::Argument::StackSpan{
+          CurrentOffset, Arguments[Index].SizeOnStack
+        };
+        CurrentOffset += Arguments[Index].SizeOnStack;
+      }
+    }
+
+    DistributedArgument RetValue = distributeReturnValue(Function.ReturnType);
+    revng_assert(RetValue.SizeOnStack == 0);
+    Result.ReturnValue.Registers = std::move(RetValue.Registers);
+
+    Result.CalleeSavedRegisters.resize(AT::CalleeSavedRegisters.size());
+    llvm::copy(AT::CalleeSavedRegisters, Result.CalleeSavedRegisters.begin());
+
+    Result.FinalStackOffset = 0; // TODO: calculate this offset properly.
```

Comment:
Mmmh, is this because we don't handle stack arguments yet?
At least set it to `model::Architecture::getCallPushSize`, which is the right thing to do for the ABIs where stack arguments cleanup responsibility is left to the caller.

Reply:
Yeah, but if we go that far, we might as well implement that properly (since we ~~have~~ had (it's easy to reintroduce) an `abi::Trait::CalleeIsResponsibleForStackCleanup` flag) and set them properly.
So, it's `getCallPushSize` + the combined size of stack arguments + (some extra stuff abi wants to write to the stack here ???) for caller-cleaned-case and what was it for callee-cleaned-case?

Reply:
Done: https://github.com/revng/revng/pull/131/commits/ebf296a3672b71363d27ecfc68cfcbfbe43a1bf2.